src/github.com/stellar/gateway/listener/payment_listener_test.go

```diff
@@ -203,22 +238,28 @@ func TestPaymentListener(t *testing.T) {
 			})
 		})
 
-
-
 		Convey("When unable to load transaction memo", func() {
 			operation.Type = "payment"
 			operation.To = "GATKP6ZQM5CSLECPMTAC5226PE367QALCPM6AFHTSULPPZMT62OOPMQB"
 			operation.AssetCode = "USD"
 			operation.AssetIssuer = "GD4I7AFSLZGTDL34TQLWJOM2NHLIIOEKD5RHHZUW54HERBLSIRKUOXRR"
 
 			mockRepository.On("GetReceivedPaymentByOperationID", int64(1)).Return(nil, nil).Once()
+
+			mockEntityManager.On("Persist", mock.AnythingOfType("*entities.ReceivedPayment")).
+				Run(ensurePaymentStatus(t, operation, "Processing...")).Return(nil).Once()
+
+			mockEntityManager.On("Persist", mock.AnythingOfType("*entities.ReceivedPayment")).
+				Run(ensurePaymentStatus(t, operation, "Unable to load transaction memo")).Return(nil).Once()
+
 			mockHorizon.On("LoadMemo", &operation).Return(errors.New("Connection error")).Once()
 
 			Convey("it should return error", func() {
 				err := paymentListener.onPayment(operation)
-				assert.Error(t, err)
+				assert.NoError(t, err)
```

Comment:
The name of this section is "it should return an error" but then you change the assertion to `NoError`.